language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java

```diff
@@ -78,11 +81,15 @@
                 }
             });
         }
-
+        Project project = ctx.workspace().project(ctx.filePath()).orElseThrow();
+        Path projectRoot = (project.kind() == ProjectKind.SINGLE_FILE_PROJECT)
+                ? project.sourceRoot().getParent() :
+                project.sourceRoot();
         // Get available diagnostics based code-actions
         ctx.allDiagnostics().stream().
-                filter(diag -> CommonUtil.isWithinRange(ctx.cursorPosition(),
-                                                        CommonUtil.toRange(diag.location().lineRange())))
+                filter(diag -> projectRoot.resolve(diag.location().lineRange().filePath()).equals(ctx.filePath()) &&
+                        CommonUtil.isWithinRange(ctx.cursorPosition(), CommonUtil.toRange(diag.location().lineRange()))
```

Comment:
While I was trying to implement a fix for this, I noticed that the `ctx.filePath()` refers to the absolute path while the `diagnostic.location().lineRange().filePath()` refers to just file name. This will cause the above equals check to fail. 

We can use something like below instead:
```
        context.filePath().endsWith(diagnostic.location().lineRange().filePath())
```
![Screenshot from 2021-02-16 08-37-52](https://user-images.githubusercontent.com/11285165/108013913-bdcdf800-7032-11eb-8d58-c0d9417b8edb.png)

Reply:
@IMS94 I didn't get any issues when testing this in a project, anyway will have a look. `endsWith()` is a string matching and has some more weird behaviors such as matching with other module's file with the same name(diag location.filePath() contains the relative path only).

Reply:
@rasika You are correct. I missed the `projectRoot.resolve()` part you have used. No issue with this then.